### PR TITLE
Update to metababel 1.0.0 interface

### PR DESCRIPTION
**Pull request by DonAurelio**
We updated the `register_callbacks_initialize/finalize_usr_data` to `register_callbacks_initialize/finalize_component`

### CODE REVIEW

**Comment by TApplencourt** [comment]
We should add in the curernt master `>=0.8 and < 1.0` as we will break compability

**Comment by DonAurelio** [comment]
We are now good to go. The CI will still fail since the last version of the metababel gem is [0.1.0](https://rubygems.org/gems/metababel/versions/0.1.0).

File: configure.ac [hunk 1]
```diff
@@ -70,7 +70,8 @@ PKG_CHECK_MODULES([PROTOBUF], [protobuf >= 3.0])
 AX_RUBY_EXTENSION([cast-to-yaml], [yes])
 AX_RUBY_EXTENSION([nokogiri], [yes])
 AX_RUBY_EXTENSION([babeltrace2], [yes])
-AX_RUBY_EXTENSION([metababel >= 0.0.5], [yes])
+AX_RUBY_EXTENSION([metababel >= 0.8.0], [yes])
+AX_RUBY_EXTENSION([metababel < 1.0.0], [yes])
```
Comment (TApplencourt) [inline configure.ac:74]:
```suggestion
AX_RUBY_EXTENSION([metababel >= 1.0.0], [yes])
```

File: xprof/btx_aggreg.cpp [hunk 2]
```diff
@@ -17,11 +17,16 @@ struct aggreg_s {
 };
 using aggreg_t = struct aggreg_s;
 
-static void initialize_usr_data_callback(void *btx_handle, void **usr_data) {
+static void initialize_component_callback(void *btx_handle, void **usr_data) {
```
Comment (TApplencourt) [inline xprof/btx_aggreg.cpp:20]:
```suggestion
static void initialize_component_callback(void **usr_data) {
```

**Review by TApplencourt — APPROVED** [review]

**Pull request merged**